src/core/thread/mesh_forwarder.cpp

```diff
@@ -888,7 +901,10 @@ void MeshForwarder::SetPollPeriod(uint32_t aPeriod)
             mPollPeriod = aPeriod;
         }
 
-        mPollTimer.Start(mPollPeriod);
+        if (mPollTimer.IsRunning() && ((mNetif.GetMle().GetDeviceMode() & Mle::ModeTlv::kModeFFD) == 0))
```

Comment:
I am not sure if  the `mPollTimer.IsRunning()` check is needed here. What about the case where we get here for first time and poll timer is not running, with this then the poll timer won't start...